soc/intel/intel_adsp/ace/power.c

```diff
@@ -135,7 +135,13 @@ static ALWAYS_INLINE void _save_core_context(uint32_t core_id)
 	core_desc[core_id].thread_ptr = XTENSA_RUR("THREADPTR");
 	__asm__ volatile("mov %0, a0" : "=r"(core_desc[core_id].a0));
 	__asm__ volatile("mov %0, a1" : "=r"(core_desc[core_id].a1));
+
+#if CONFIG_MP_MAX_NUM_CPUS == 1
+	/* With one core only, the memory is mapped in cache and we need to flush
+	 * it.
```

Comment:
Actually I'm pretty sure this is needed.  At least on early cAVS platforms, the linker script would only do the memory space separation on multicore setups, leaving .bss data like core_desc[] cached.  Though not sure if that was MP_MAX_NUM_CPUS or SMP or what.

Basically the comment sounds to me like someone bumped into this as a bug and fixed it?

Reply:
it was always flushing it, but this memory is not in cached region ... unless we have only one core.
Basically, I had removed this flush but when we built with only one core the memory is mapped in cache and we need it.

The change is basically doing, call this cache flush *if* we build with only one core.